5_evaluate/src/eval_utility_fxns.R

```diff
@@ -71,36 +71,42 @@ match_pred_obs <- function(preds_file, eval_obs) {
       mutate(obs_1d, pred = NA)
     })
     return(interp_1d)
-  }))
+  })) %>%
+    filter(!is.na(pred)) # Filter out rows where there are no matching predictions for observed depths (e.g. observed depth exceeds predicted depth)
   
   return(pred_obs)
 }
 
 #' @title prep pred-obs data for evaluation
 #' @description add grouping variables to matched pred-obs for evaluation.
 #' @param pred_obs the tibble of matched observations and predictions
-#' @param surface_cutoff_depth maximum depth for which predictions are
+#' @param lake_depths tibble of lake depths for evluation sites
+#' @param surface_max_depth maximum depth for which predictions are
 #' considered to be in the 'surface' depth class. Currently this is set
 #' as a global value, and is not lake-specific
-#' @param middle_cutoff_depth maximum depth for which predictions are
-#' considered to be in the 'middle' depth class. Currently this is set
-#' as a global value, and is not lake-specific
+#' @param bottom_depth_factor a factor indicating the proportion
+#' of the water column considered to be *outside* of the 'bottom' depth 
+#' class. When multiplied by each lake's depth, the resulting value is the 
+#' minimum depth for which predictions for each lake are considered to be 
+#' in the 'bottom' depth class
 #' @param doy_bin_size # of days to include in each doy bin
 #' @param temp_bin_size # of degrees to include in each temperature bin
 #' @return a tibble of matched predictions and observations with the 
 #' difference between the prediction and observation (pred_diff), the
 #' depth_class (surface, middle, or bottom), year, doy, doy_bin (size 
 #' set by `doy_bin_size`), and season of each pred-obs pair, and the 
 #' temperature bin (size set by `temp_bin_size`) into which each observation falls 
-prep_data_for_eval <- function(pred_obs, surface_cutoff_depth, middle_cutoff_depth, doy_bin_size, temp_bin_size) {
+prep_data_for_eval <- function(pred_obs, lake_depths, surface_max_depth, bottom_depth_factor, doy_bin_size, temp_bin_size) {
   
   eval_pred_obs <- pred_obs %>%
+    left_join(lake_depths, by='site_id') %>%
     mutate(pred_diff = pred - obs,
            depth_class = case_when(
-             depth <= surface_cutoff_depth ~ 'surface',
-             depth > surface_cutoff_depth & depth <= middle_cutoff_depth ~ 'middle',
-             TRUE ~ 'bottom'
+             depth <= surface_max_depth ~ 'surface',
+             depth >= (bottom_depth_factor * lake_depth) ~ 'bottom',
+             TRUE ~ 'middle'
            ),
+           depth_class = factor(depth_class,levels=c('surface','middle','bottom')),
```

Comment:
Ok, here is where I get an answer to my above question about how many depth classes there are currently. Makes sense.